client/src/schema/repository.ts

```diff
@@ -352,17 +356,26 @@ export class BaseClient<D extends Record<string, Repository<any>> = Record<strin
       const [field, linkTable] = link;
       const value = o[field];
 
-      if (value && typeof value === 'object') {
-        const { id } = value as any;
-        if (Object.keys(value).find((col) => col === 'id')) {
-          o[field] = this.initObject(linkTable, value);
-        } else if (id) {
+      if (isIdentifiable(value)) {
+        const { id } = value;
+
+        if (depth === -1 || depth >= 10) {
           o[field] = {
             id,
-            get: () => {
-              this.db[linkTable].read(id);
+            xata: { version: 1 },
```

Comment:
🤔  is there a case when we don’t get the version? If that’s the case I’d prefer to have here an undefined or if the field needs to be defined we can default to -1. But first, I’d like to understand why in this case we don’t get the version.

Reply:
This is the depth >= 10 case, where we have already gone too deep, and we do not retrieve from the server all the properties and leave the id + the read/update/delete methods

Reply:
```suggestion
            xata: { version: -1 },
```